crates/bevy_app/src/app.rs

```diff
@@ -619,6 +631,45 @@ impl App {
             .add_system_to_stage(CoreStage::First, Events::<T>::update_system)
     }
 
+    /// Inserts an initialization resource to the current [App] and overwrites any resource
+    /// previously added of the same type.
+    ///
+    /// An initialization resource is used at startup for plugin initialisation and configuration.
+    /// All initialization resources inserted must be consumed by a plugin and removed before the
+    /// application is ran.
+    pub fn insert_initialization_resource<T>(&mut self, resource: T) -> &mut Self
```

Comment:
The new Method names are pretty long.
Could they be terser like: `insert_init_resource`

Any other Ideas?

Reply:
Now that i tnik about it `insert_init_resource` is also bad as `init_resource` and `insert_resource` already exist 🤔

Reply:
I think that fundamentally these should not be called "initialization resources", due to the conflict with init_resource.

Perhaps "config resources" or "setup resources"?

Reply:
To me, "config" implies something that can still be used at runtime. The PR description calls them startup resources and it makes a lot of sense to me considering we already have `.add_startup_systems()` having `.add_startup_resource` feels pretty clean.

Reply:
I renamed from `add_startup_resource` because they won't be available during startup systems. I think `setup` sounds pretty nice